### PR TITLE
Fix/97 persistent notifications

### PR DESCRIPTION
**Pull request by jgoldhammer**
### Status
**READY**

### Description
The merge request should show give user hints about the notification settings for Meetingbar.
We will display a little result when the user activates the notifications 

### Steps to Test or Reproduce
See issue #97 for screenshots etc.

### CODE REVIEW

File: MeetingBar/Notifications.swift [hunk 1]
```diff
@@ -12,13 +12,20 @@ import Defaults
...
             NSLog("Access to notications granted")
+            center.getNotificationSettings { notificationSettings in
+                NSLog("Authorisation status \(notificationSettings.authorizationStatus.rawValue)")
+                NSLog("Alert Style settings \(notificationSettings.alertStyle.rawValue)")
+            }
         } else {
             NSLog("Access to notications denied")
         }
     }
+
```
Comment (leits) [inline MeetingBar/Notifications.swift:27]: Check swiftlint warning for these lines (27-28)

File: MeetingBar/Notifications.swift [hunk 2]
```diff
@@ -71,8 +91,17 @@ func scheduleEventNotification(_ event: EKEvent) {
...
+    center.add(request) { error in
+        if let error = error {
+            NSLog("%@", "request \(request) could not be added because of error \(error)")
+        } else {
+            NSLog("%@", "request \(request) was added")
+        }
+    }
 }
+
+
```
Comment (leits) [inline MeetingBar/Notifications.swift:107]: Swiftlint warning for 107.

File: MeetingBar/PreferencesView.swift [hunk 3]
```diff
@@ -641,6 +642,25 @@ struct JoinEventNotificationPicker: View {
     @Default(.joinEventNotification) var joinEventNotification
     @Default(.joinEventNotificationTime) var joinEventNotificationTime
 
+    func checkNotificationSettings() -> (Bool, Bool) {
+        var noAlertStyle: Bool = false
+        var notificationsDisabled: Bool = false
+
+        let center = UNUserNotificationCenter.current()
+        let group = DispatchGroup()
```
Comment (leits) [inline MeetingBar/PreferencesView.swift:650]: Great way. Didn't know about DispatchGroup with wait feature.

File: MeetingBar/PreferencesView.swift [hunk 4]
```diff
@@ -651,6 +671,19 @@ struct JoinEventNotificationPicker: View {
...
+        if noAlertStyle && !disabled && joinEventNotification {
+            Text("⚠️ Your macos notification settings for Meetingbar are currently not set to alert. Please activate alerts if you want to have persistent notifications.").foregroundColor(Color.gray).font(.system(size: 12))
+        }
+
+        if disabled && joinEventNotification {
+            Text("⚠️ Your macos notification settings for Meetingbar are currently off. Please enable the notifications in macos system settings to do not miss a meeting.").foregroundColor(Color.gray).font(.system(size: 12))
+        }
+
+        if !disabled && !noAlertStyle && joinEventNotification {
+            Text("🎉 Your macos notification settings for Meetingbar are enabled with alert style.").foregroundColor(Color.gray).font(.system(size: 12))
```
Comment (leits) [inline MeetingBar/PreferencesView.swift:685]: I think we may not show anything to the user when everything is fine.

Comment (jgoldhammer) [reply MeetingBar/PreferencesView.swift:685]: Ok, I will remove it.

File: MeetingBar/Notifications.swift (hunk 1 above)
Comment (jgoldhammer) [reply MeetingBar/Notifications.swift]: Fixed

**Review by leits — APPROVED** [review]

**Pull request merged**